internal_worker_base.go

```diff
@@ -121,18 +133,128 @@ func (bw *baseWorker) Start() {
 		return
 	}
 
-	// Launch the routines to do work
-	for i := 0; i < bw.options.routineCount; i++ {
+	for i := 0; i < bw.options.pollerCount; i++ {
 		bw.shutdownWG.Add(1)
-		go bw.execute(i)
+		go bw.runPoller()
 	}
+	bw.shutdownWG.Add(1)
+	go bw.runTaskDispatcher()
 
 	bw.isWorkerStarted = true
 	traceLog(func() {
-		bw.logger.Info("Started Worker", zap.Int("RoutineCount", bw.options.routineCount))
+		bw.logger.Info("Started Worker",
+			zap.Int("PollerCount", bw.options.pollerCount),
+			zap.Int("MaxConcurrentTask", bw.options.maxConcurrentTask),
+			zap.Int("MaxTaskRps", bw.options.maxTaskRps),
+		)
 	})
 }
 
+func (bw *baseWorker) isShutdown() bool {
+	select {
+	case <-bw.shutdownCh:
+		return true
+	default:
+		return false
+	}
+}
+
+func (bw *baseWorker) runPoller() {
+	defer bw.shutdownWG.Done()
+
+	bw.pollerRequestCh <- struct{}{} // initial poll request to kick off the poller loop
```

Comment:
if pollLatency=5ms, two poller is able to produce 400 task per second, which will keep all worker busy. The dispatcher is only receiving from the task queue, it is not the worker. Once a worker starts, it checks if the concurrent task count is less than max, and if so, it sends poller request before process the task.